src/app/domains/transactions/domains.transactions.gateway.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { WalletsDetailsTransaction } from 'src/app/pages/wallets/wallet-details/pages-wallet-details-item.model';
+import { ITransactionPayload, IWalletTransactionApiResponse } from './domains.transactions.types';
+import { TServerEntityId } from 'src/app/common/http/common.http.types';
+import { API_TRANSACTIONS_URL } from './domains.transactions.constants';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DomainsTransactionsGateway {
+  public constructor(
+    private readonly http: HttpClient,
+  ) {
+  }
+
+  public getWalletTransactions(id: TServerEntityId): Observable<IWalletTransactionApiResponse[]> {
+    return this.http.get<IWalletTransactionApiResponse[]>(API_TRANSACTIONS_URL + `?walletId=${id}`);
```

Comment:
I think `http.get` has `params` property that can be used to provide `walletId` in nicer way